### ISSUE TITLE
Questions, Changing # vcpu's - iohyve snapshot - uuid - import VM

### ISSUE THREAD

**Issue by pobkac** [pr1ntf/iohyve#196]
### Hello, a couple of more questions.  The OS is FreeBSD 11.0-ALPHA6 (GENERIC) #0 r302303 with bhyve updated through the git UEFI-GOP
# 

1/. 

I can't seem to change the number of vcpu's.  In fact I installed windows with cpu=4 but only 2 cores are reported inside win10.  I did this as previously I wasn't able to increase the number of cores.
I am using a asrock c2750 motherboard and with only 2 vcpu's win10 isn't really usable.  I need to do more tests but copying the files between the virtual cd and virtual hard drive gave approximately 5-8 MBytes/s and seemed to be cpu limited.  Copying over the network from another windows machine gave 10-20 MB/s.  Copying straight to FreeBSD gives 50+ MB/s. 
ioh
root@11alpha6:/iohyve/Firmware # iohyve getall win10a
Getting win10a iohyve properties...
bargs          -H_-w
boot           0
con            nmdm0
cpu            4
description    Mon Jul 11 21:58:42 ACST 2016
fw             BHYVE_UEFI.fd
loader         uefi
name           win10a
os             default
persist        1
ram            4096M
size           50G
tap            tap1
template       NO
vnc            YES
vnc_h          1024
vnc_ip         192.168.1.30
vnc_port       5910
vnc_tablet     YES
vnc_w          1280
vnc_wait       no
# 

2/. I have also tried to run the installed vm directly in bhyve but with out success to try and see where the limitation is.
bhyve -w -H -c 4 -m 4096 \
    -s 0,hostbridge \
    -s 3,ahci-cd,/iohyve/ISO/null.iso \
    -s 4,ahci-hd,/dev/zvol/sam830/iohyve/win10a/disk0 \
    -s 5,virtio-net,tap1 -l bootrom,/iohyve/Firmware/BHYVE_UEFI_20160526.fd \
    -s 29,fbuf,tcp=192.168.1.30:5910,w=1280,h=1024 -s 30,xhci,tablet \
    -s 31,lpc -l com1,stdio \
    win10a
The error  is "Invalid bootrom size 3"  I have tried both the BHYVE_UEFI.fd created below and the downloaded BHYVE_UEFI_20160526.fd
Are you able to give any insight into the error comment?

The BHYVE_UEFI.fd I built locally on the FreeBSD machine hosting bhyve in case there was some compatibility advantages.  FWIW the freebsd instructions didn't work on my alpha5 install, as the make file wasn't present in /usr/ports
# 

3/. I am having an issue cloning a vm, but the iohyve snapshot command works.

root@11alpha6:/iohyve/Firmware # iohyve clone win10a win10b
Not a valid guest name
root@11alpha6:/iohyve/Firmware # iohyve clone win10a /TankC/FreeBSD/win10b
Not a valid guest name
the win10a is a valid vmm as shown above by the iohyve getall win10a.
I tried this before and after destroying the vm, the vm was also stopped prior to trying.
# 

4/. is there any way to extract the uuid from iohyve/bhyve?  After reading sites like http://www.zdnet.com/article/windows-activation-trips-up-virtual-machine-clones-even-on-same-system/ I'm keen to avoid such issues in future.  I'm not trying to run multiple copies of the same windows license but just want ensure a smooth transition from replacing alpha 6 with 10.3 and upgrading to 11.0-release at the appropriate time.  Are there any restrictions on the uuid?  I.E, can it be any "random" string or human readable descritption or does it need to be numerical? 
I assume that I can set the mac using iohyve set win10a tap=tap1,mac=xx:xx:xx:xx:xx:xx
# 

5/. you have an export command, but is there a corresponding import command?
Again many thanks
Brendan
edited to remove funky formatting

**Comment by pr1ntf** [comment]
I'll get to your questions, possibly separating them into their own issues, but can you fix some formatting? The above is kind of hard on the eyes. If you click "Edit" in the GitHub user interface, you can use they're fancy editor to preview your formatting.

**Comment by pobkac** [comment]
hmm, yuk. Sorry about that, that is pretty dam ugly!!  I copied the text in from a text editor.  No idea where the formatting came from, I'll blame Windows, rather then the operator :).  I am trying to ween myself off Windoze....

**Comment by pr1ntf** [comment]
No worries, we'll try to ween you off of Windows :wink: 
1. After you gave the VM more vCPU's, did you restart the VM? Changes to memory and CPU's don't take effect until after the VM is **destroyed** (`iohyve destroy vmname`) and started again. If this continues to be a problem, please open a separate issue. 
2. That sounds like a problem with your bootrom file, not iohyve, but you may want to put all of your LPC devices together (like after your `-s 31` ...`). You may want to ask in the #bhyve channel on Freenode or ask the mailing list. 
3. Check the [man page](https://github.com/pr1ntf/iohyve/blob/master/iohyve.8.txt#L263) and try using `-r`. There seems to be a bug in the clone function at the moment that does not allow for carbon copy cloning. 
4. There is a way to pass UUID to bhyve, but that is not supported in iohyve at this time. If you would like this feature, please open a separate issue as a "feature request." Also, check the [man page](https://github.com/pr1ntf/iohyve/blob/master/iohyve.8.txt#L196) on how to set MAC addresses. 
5. There is currently no `import` function, but it is [planned](https://github.com/pr1ntf/iohyve/wiki/General-Path-to-1.0#079-goals). Some other things need to happen first. 

I am closing out this issue, but if you continue to have problems, please open a separate issue for each problem or request. Thank you.
